libarchive/test/test_write_format_xar.c

```diff
@@ -274,6 +306,13 @@ test_xar(const char *option)
 
 DEFINE_TEST(test_write_format_xar)
 {
+	/* xar mandates the use of UTF-8 XML; if we cannot
+	 * use UTF-8, perhaps we should not write xar. */
+	if (NULL == setlocale(LC_ALL, "en_US.UTF-8")) {
+		skipping("en_US.UTF-8 locale not available on this system.");
```

Comment:
What are the visible side-effects of not using that locale? Should we move the actual set/check locale in the library instead?

Why aren't we checking other UTF-8 locates - be that `en_gb.UTF-8`, `ko.UTF-8` ... or the recently introduced `C.UTF-8` one? The last one might be Linux and/or glibc specific.

Reply:
That's a really good question! I don't think the library takes a position on whether _formats_ work if certain encodings aren't available... but it sure does take a stance on whether tests can be run!

```
test_zip_filename_encoding.c:   if (NULL == setlocale(LC_ALL, "en_US.UTF-8")) {
test_read_format_zip_7075_utf8_paths.c: if (NULL == setlocale(LC_ALL, "en_US.UTF-8")) {
test_read_format_ustar_filename.c:      if (NULL == setlocale(LC_ALL, "en_US.UTF-8")) {
test_read_format_ustar_filename.c:      if (NULL == setlocale(LC_ALL, "en_US.UTF-8")) {
test_read_format_ustar_filename.c:      if (NULL == setlocale(LC_ALL, "en_US.UTF-8")) {
test_read_format_tar_filename.c:        if (NULL == setlocale(LC_ALL, "en_US.UTF-8")) {
test_read_format_rar.c:  if (NULL == setlocale(LC_ALL, "en_US.UTF-8")) {
test_read_format_lha_filename_utf16.c:  if (NULL == setlocale(LC_ALL, "en_US.UTF-8")) {
```

Reply:
I wonder if using `export LC_ALL=....` in the test suite runners (cmake/autotools) won't be a more scalable and consistent experience?

Either way - having your comment in the commit message, is great for posterity et al.